mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java

```diff
@@ -216,42 +219,103 @@ private synchronized void closeCreatedStores() {
     /**
      * Checks the directory configured exists and is readable and writeable, or creates it
      * if it does not exist yet.
+     *
+     * @param path the path to check
+     * @param type metadata store type
      */
-    private void checkDir(File dir, String type) {
-        if (!dir.exists()) {
+    private void checkDir(Path path, String type) {
+        if (!Files.exists(path)) {
             try {
-                boolean created = dir.mkdirs();
+                boolean created = Files.exists(Files.createDirectories(path));
                 if (!created) {
-                    String msg = "Unable to create metadata store " + type + " directory " + dir.getAbsolutePath();
+                    String msg = "Unable to create metadata store " + type + " directory " + path.toAbsolutePath();
                     throw new MCRConfigurationException(msg);
                 }
             } catch (Exception ex) {
-                String msg = "Exception while creating metadata store " + type + " directory " + dir.getAbsolutePath();
+                String msg = "Exception while creating metadata store " + type + " directory " + path.toAbsolutePath();
                 throw new MCRConfigurationException(msg, ex);
             }
         } else {
-            if (!dir.canRead()) {
-                String msg = "Metadata store " + type + " directory " + dir.getAbsolutePath() + " is not readable";
+            if (!Files.isReadable(path)) {
+                String msg = "Metadata store " + type + " directory " + path.toAbsolutePath() + " is not readable";
                 throw new MCRConfigurationException(msg);
             }
-            if (!dir.canWrite()) {
-                String msg = "Metadata store " + type + " directory " + dir.getAbsolutePath() + " is not writeable";
+            if (!Files.isWritable(path)) {
+                String msg = "Metadata store " + type + " directory " + path.toAbsolutePath() + " is not writeable";
                 throw new MCRConfigurationException(msg);
             }
-            if (!dir.isDirectory()) {
-                String msg = "Metadata store " + type + " " + dir.getAbsolutePath() + " is a file, not a directory";
+            if (!Files.isDirectory(path)) {
```

Comment:
isDirectory should be checked first for other "directory" exceptions listed before...